esp-hal/src/gpio/mod.rs

```diff
@@ -1530,263 +1562,12 @@ impl<'d> Input<'d> {
     pub fn into_peripheral_output(self) -> interconnect::OutputSignal {
         self.pin.into_peripheral_output()
     }
-}
-
-/// Open-drain output pin configuration.
-#[cfg_attr(feature = "defmt", derive(defmt::Format))]
-#[derive(Debug, Clone, Copy, PartialEq, Eq, procmacros::BuilderLite)]
-#[non_exhaustive]
-pub struct OutputOpenDrainConfig {
-    /// Initial output level of the pin.
-    pub level: Level,
-    /// Initial pull of the pin.
-    pub pull: Pull,
-}
-
-impl Default for OutputOpenDrainConfig {
-    fn default() -> Self {
-        Self {
-            level: Level::Low,
-            pull: Pull::None,
-        }
-    }
-}
-
-/// Open drain digital output.
-///
-/// This driver configures the GPIO pin to be an open drain output driver.
```

Comment:
I think this comment is still valuable, maybe we can massage it into the top level docs? It would be nice to have something in the top level docs about how to get an open drain output pin.

Reply:
Thanks, also the Output docs were incorrect, so I sent in another commit to tweak docs a bit